File: examples/canonical-faker/go/internal/profiles/profiles.go
package profiles

type Profile struct {
	Name     string
	NumOrgs  int
	NumUsers int
	Seed     int
}

var (
	Smoke  = Profile{Name: "smoke", NumOrgs: 3, NumUsers: 10, Seed: 42}
	Small  = Profile{Name: "small", NumOrgs: 20, NumUsers: 200, Seed: 1000}
	Medium = Profile{Name: "medium", NumOrgs: 200, NumUsers: 5000, Seed: 5000}
	Large  = Profile{Name: "large", NumOrgs: 2000, NumUsers: 50000, Seed: 9999}
)

func Resolve(name string, seedOverride int) Profile {
	var profile Profile
	switch name {
	case "small":
		profile = Small
	case "medium":
		profile = Medium
	case "large":
		profile = Large
	default:
		profile = Smoke
	}
	if seedOverride != 0 {
		profile.Seed = seedOverride
	}
	return profile
}

File: examples/canonical-faker/elixir/lib/tenant_core/profiles.ex
defmodule TenantCore.Profile do
  defstruct [:name, :seed, :num_orgs, :num_users]

  def smoke do
    %__MODULE__{name: "smoke", seed: 42, num_orgs: 3, num_users: 10}
  end

  def small do
    %__MODULE__{name: "small", seed: 42, num_orgs: 10, num_users: 50}
  end

  def medium do
    %__MODULE__{name: "medium", seed: 42, num_orgs: 100, num_users: 500}
  end

  def large do
    %__MODULE__{name: "large", seed: 42, num_orgs: 1000, num_users: 5000}
  end
end
